File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> res;
        map<int,int> mp;
        for(int x:nums){
            mp[x]++;
        }
        int mx=-1;
        vector<pair<int,int>> v(mp.begin(), mp.end());
        sort(v.begin(), v.end(), [](auto &a, auto &b) {
            return a.second > b.second;
        });
        int j=1;
        for(auto p:v){
            res.push_back(p.first);
            j++;
            if(j>k){
                break;
            }
        }
        return res;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int n=nums.length;
        int [] res= new int[k];
        HashMap<Integer,Integer> 지도=new HashMap<>();
        for(int num:nums){
            지도.put(num,지도.getOrDefault(num,0)+1);
        }
        List<Map.Entry<Integer,Integer>> list=new ArrayList<>(지도.entrySet());
        list.sort((a,b)->a.getValue()-b.getValue());
        int idx=0;
        for(int i=list.size()-1;i>=list.size()-k;i--){
            res[idx++]=list.get(i).getKey();
        }
        return res;
    }
}